src/api/Alerts/types.ts

```diff
@@ -0,0 +1,187 @@
+/**
+ * Copyright Linkurious SAS 2012 - 2019
+ *
+ * - Created on 2019-10-30.
+ */
+
+import {FolderChildren, IDataSourceParams, PersistedItem} from '../commonTypes';
+import {GraphQueryDialect} from '../GraphQuery';
+import {LkEdge, LkNode} from '../graphItemTypes';
+
+export enum ColumnType {
+  STRING = 'string',
+  NUMBER = 'number'
+}
+
+export interface ICreateAlertParams extends IDataSourceParams {
+  title: string;
+  query: string;
+  dialect: GraphQueryDialect;
+  folder?: number;
+  enabled: boolean;
+  columns: Array<{
+    type: ColumnType;
+    columnName: string;
+    columnTitle: string;
+  }>;
+  cron: string;
+  matchTTL?: number;
+  maxMatches?: number;
+}
+
+export interface Alert extends PersistedItem {
+  title: string;
+  sourceKey: string;
+  query: string;
+  dialect: GraphQueryDialect;
+  folder: number;
+  enabled: boolean;
+  columns: Array<{
+    type: ColumnType;
+    columnName: string;
+    columnTitle: string;
+  }>;
+  cron: string;
+  matchTTL: number;
+  maxMatches: number;
+  userId: number;
+  lastRun: string;
+  lastRunProblem?: {
+    error: string;
+    partial: boolean;
+  };
+  nextRun?: string;
+}
+
+export interface CreateAlertResponse extends Alert {}
+
+export interface IUpdateAlertParams extends Partial<ICreateAlertParams> {
+  id: number;
+}
+
+export interface UpdateAlertResponse extends Alert {}
+
+export interface IDeleteAlertParams extends IDataSourceParams {
+  id: number;
+}
+
+export interface ICreateAlertFolderParams extends IDataSourceParams {
+  title: string;
+}
+
+// TODO add sourceKey to server
+export interface AlertFolder extends PersistedItem {
+  title: string;
+  parent: number;
+  sourceKey: string;
+}
+
+export interface CreateAlertFolderResponse extends AlertFolder {}
+
+export interface IUpdateAlertFolderParams extends IDataSourceParams {
+  id: number;
+  title: string;
+}
+
+// TODO return alert folder in server
+export interface UpdateAlertFolderResponse extends AlertFolder {}
+
+export interface IDeleteAlertFolderParams extends IDataSourceParams {
+  id: number;
+}
+
+export interface IGetAlertTreeParams extends IDataSourceParams {}
+
+export interface GetAlertTreeResponse {
+  id: -1;
+  title: 'root';
+  type: 'folder';
+  children: FolderChildren<Alert, 'alert'>;
+}
+
+export interface IGetAlertParams extends IDataSourceParams {
+  id: number;
+}
+
+export interface GetAlertResponse extends Alert {}
+
+export interface IGetMatchParams extends IDataSourceParams {
+  alertId: number;
+  matchId: number;
+}
+
+export type MatchStatus = 'unconfirmed' | 'confirmed' | 'dismissed';
+
+export interface Match extends PersistedItem {
+  sourceKey: string;
+  alertId: number;
+  hash: string;
+  status: MatchStatus;
+  user: {
+    id: number;
+    username: string;
+    email: string;
+  };
+  viewers: {
```

Comment:
`viewers` is plural but it contains only one user reference.

Reply:
because it's an array of those :D